Get zha switch and binary_sensor state on HA start

SteveEasley opened the pull request:
## Description:

Ensures the zha switch and binary_sensor platform reads the current state of the switch on HA startup. This mirrors how the zha light works.

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

maxlaverse committed:
Fix webdav calendar schema (#11185)

jannau committed:
homematic: add username and password to interface config schema (#11214)

Fixes #11191, the json-rpc name resolving method requires user account
and password.

OverloadUT committed:
Fix detection of if a negative node is in use (#11255)

* Fix detection of if a negative node is in use

Fix a problem where every negative node gets detected as in-use. Code was not checking the correct property.

* Allow protected access

pvizeli committed:
Bugfix homematic available modus (#11256)

CTLS committed:
Fix inverted sensors on the concord232 binary sensor component (#11261)

* Fix inverted sensors on the concord232 binary sensor component

* Changed from == Tripped to != Normal

ziotibia81 committed:
Fix handling zero values for state_on/state_off (#11264)

veleek committed:
Fix async IO in Sesame lock component. (#11054)

* Call update on Sesame devices to cache initial state

* Switch to using async_add_devices

* Fix line length

* Fix Lint errors

* Fix more Lint errors

* Cache pysesame properties

* Updates from CR feedback

maxlaverse committed:
Fix allday events in custom_calendars (#11272)

OddBloke committed:
Fix webostv select source (#11227)

* Fix reuse of variable name

This should fix #11224.

* Add tests for LgWebOSDevice.select_source

andreacampi committed:
Support multiple Hue bridges with lights of the same id (#11259)

* Improve support for multiple Hue bridges with lights that have the same id.

The old code pre-refactoring kept a per-bridge list of lights in a closure; my refactoring moved that to hass.data, which is convenient but caused them to conflict with each other.

Fixes #11183

* Update test_hue.py

rwa committed:
Fix unpredictable entity names in concord232 binary_sensor (#11292)

OverloadUT committed:
Fix leak sensors always showing Unknown until Wet (#11313)

Leak sensors were using the "wet" node as a negative node, which prevented them from ever gettng a Dry status unless the user pressed the button on the hardware after every Hass reboot.

This change ignores the Wet node, as it is just always the exact inverse of the Dry node. We don't need to watch both.

pvizeli committed:
Don't block on sevice call for alexa (#11358)

* Don't block on sevice call for alexa

* fix tests

Commit:
Get zha switch state on HA start

Commit:
Get zha switch state on HA start

akatrevorjay committed:
Account for User-Agent being non-existent, causing a TypeError (#11064)

* Account for User-Agent being non-existent, causing a TypeError

* Actually fix case of no user-agent with last resort

* Return es5 as last resort

* Update __init__.py

* Update __init__.py

Gamester17 commented:
@SteveEasley Nice work! Did you by the way also notice pull request #11305 by @dmulcahey

dmulcahey commented:
@Gamester17 I closed mine at the moment so I can come up with a better way to determine units of measure.

balloob committed:
Cloud Updates (#11404)

* Verify stored keys on startup

* Handle Google Assistant messages

* Fix tests

* Don't verify expiration when getting claims

* Remove email based check

* Lint

* Lint

* Lint

dfinlay committed:
Add support for the renaming of Yamaha Receiver Zones via configuration file. Added a test to cover the change, plus previously untested options. (#11402)

ryanm101 committed:
Plex api update (#11423)

* Updated plexapi to 3.0.5

* Removed removed server obj that was being used for the media URL generation as .url() is available in self._session
seasons() no longer exists -> renamed to season()
season() returns obj not list so corrected
season.index returns int, cast to str to zerofill

* Fixed Linting

daenny committed:
fix: hmip-etrv-2 now working with homeassistant (#11175)

* fix: hmip-etrv-2 now working with homeassistant (see also pull request at pyhomematic)

* fix linting issue and typo

* address comment @pvizeli

* Only use cached data in current operation mode

* fix linting issue

Commit:
Fix errors in zigbee push state (#11386)

* Fix errors in zigbee push state

* Fix formatting

* Added comment

andrey-git committed:
Add on/off supported feature to climate (#11379)

* Add on/off supported feature to climate

* Lint

cdce8p committed:
Fix CONF_FRIENDLY_NAME (#11438)

andrey-git committed:
Climate: fix missing "|" (#11441)

* Add on/off supported feature to climate

* Lint

* Fix missing |

* lint

balloob committed:
Google Assistant -> Google Actions (#11442)

rofrantz committed:
Added new climate component from Daikin (#10983)

* Added Daikin climate component

* Fixed tox & hound

* Place up the REQUIREMENTS var

* Update .coveragerc

* Removed unused customization

* Prevent setting invalid operation state

* Fixed hound

* Small refactor according to code review

* Fixed latest code review comments

* Used host instead of ip_address

* No real change

* No real change

* Fixed lint errors

* More pylint fixes

* Shush Hound

* Applied suggested changes for temperature & humidity settings

* Fixed hound

* Fixed upper case texts

* Fixed hound

* Fixed hound

* Fixed hound

* Removed humidity since even the device has the feature it cant be set from API

* Code review requested changes

* Fixed hound

* Fixed hound

* Trigger update after adding device

* Added Daikin sensors

* Fixed hound

* Fixed hound

* Fixed travis

* Fixed hound

* Fixed hound

* Fixed travis

* Fixed coverage decrease issue

* Do less API calls and fixed Travis failures

* Distributed code from platform to climate and sensor componenets

* Rename sensor state to device_attribute

* Fixed hound

* Updated requirements

* Simplified code

* Implemented requested changes

* Forgot one change

* Don't allow customizing temperature unit and take it from hass (FOR NOW)

* Additional code review changes applied

* Condensed import even more

* Simplify condition check

* Reordered imports

* Disabled autodiscovery FOR NOW :(

* Give more suggestive names to sensors

SteveEasley commented:
@bacumings Likely so. I will try to implement and test it today. Hopefully get it into this PR before review.

bacumings commented:
Awesome! Thanks @SteveEasely!

Commit:
Get zha binary_switch state on HA start

Commit:
Get zha binary_switch state on HA start

ciotlosm committed:
Updated generic thermostat to respect operation_mode and added away mode (#11445)

* Updated generic thermostat to respect operation_mode and added away mode

* Updated tests to include away mode and corrected init problem for sensor state
Added more tests to improve coverage and corrected again lint errors
Fixed new test by moving to correct package
Fixed bug not restoring away mode on restart

* Added support for idle on interface through state

* Added back initial_operation_mode and modified away_temp to be only one for now

* Fixed houndci-bot errors

* Added back check for None on restore temperature

* Fixed failing tests as well

* Removed unused definitions from tests

* Added use case for no initial temperature and no previously saved temperature

Commit:
Lint fix

Commit:
Lint fix

Commit:
File permission revert

Commit:
File permission revert

SteveEasley commented:
@bacumings I have added the change to this PR. But I actually dont have a binary sensor to test with. Can you (or anyone else) test by dropping the [updated binary_sensor/zha.py file] into your `.homeassistant/custom_components/binary_sensor` folder (create the last binary_sensor folder if it does not exist)? If that's greek to you, don't worry about it, hopefully someone else can test.

tschmidty69 commented:
with this binary_sensory component
```
Jan  4 14:04:07 home hass[26095]: #033[31m2018-01-04 14:04:07 ERROR (MainThread) [homeassistant.components.binary_sensor] Error on device update!#033[0m
Jan  4 14:04:07 home hass[26095]: Traceback (most recent call last):
Jan  4 14:04:07 home hass[26095]:   File "/home/homeassistant/.homeassistant/custom_components/binary_sensor/zha.py", line 106, in safe_read
Jan  4 14:04:07 home hass[26095]:     allow_cache=False,
Jan  4 14:04:07 home hass[26095]:   File "/srv/homeassistant/lib/python3.6/site-packages/bellows/zigbee/zcl/__init__.py", line 192, in read_attributes
Jan  4 14:04:07 home hass[26095]:     result = yield from self.read_attributes_raw(to_read)
Jan  4 14:04:07 home hass[26095]:   File "/srv/homeassistant/lib/python3.6/site-packages/bellows/zigbee/zcl/__init__.py", line 159, in read_attributes_raw
Jan  4 14:04:07 home hass[26095]:     v = yield from self.request(True, 0x00, schema, attributes)
Jan  4 14:04:07 home hass[26095]:   File "/srv/homeassistant/lib/python3.6/site-packages/bellows/zigbee/application.py", line 284, in request
Jan  4 14:04:07 home hass[26095]:     v = yield from send_fut
Jan  4 14:04:07 home hass[26095]:   File "/usr/lib/python3.6/asyncio/futures.py", line 332, in __iter__
Jan  4 14:04:07 home hass[26095]:     yield self  # This tells Task to wait for completion.
Jan  4 14:04:07 home hass[26095]:   File "/usr/lib/python3.6/asyncio/tasks.py", line 250, in _wakeup
Jan  4 14:04:07 home hass[26095]:     future.result()
Jan  4 14:04:07 home hass[26095]:   File "/usr/lib/python3.6/asyncio/futures.py", line 245, in result
Jan  4 14:04:07 home hass[26095]:     raise self._exception
Jan  4 14:04:07 home hass[26095]: bellows.zigbee.exceptions.DeliveryError: Message send failure: EmberStatus.DELIVERY_FAILED
Jan  4 14:04:07 home hass[26095]: During handling of the above exception, another exception occurred:
Jan  4 14:04:07 home hass[26095]: Traceback (most recent call last):
Jan  4 14:04:07 home hass[26095]:   File "/srv/homeassistant/lib/python3.6/site-packages/homeassistant/helpers/entity_component.py", line 217, in async_add_entity
Jan  4 14:04:07 home hass[26095]:     yield from entity.async_device_update(warning=False)
Jan  4 14:04:07 home hass[26095]:   File "/srv/homeassistant/lib/python3.6/site-packages/homeassistant/helpers/entity.py", line 304, in async_device_update
Jan  4 14:04:07 home hass[26095]:     yield from self.async_update()
Jan  4 14:04:07 home hass[26095]:   File "/home/homeassistant/.homeassistant/custom_components/binary_sensor/zha.py", line 113, in async_update
Jan  4 14:04:07 home hass[26095]:     self._state = result.get('zone_status', self._state) & 3
Jan  4 14:04:07 home hass[26095]: TypeError: unsupported operand type(s) for &: 'str' and 'int'#033[0m
```

PhracturedBlue committed:
Reconnect to alarmdecoder on disconnect (#11383)

* Reconnect to alarmdecoder on disconnect

* Use track_point_in_time instead of call_later

* use alarmdecoder 1.13.2 which has a more robust reconnection fix

Commit:
Ensure state ignores unknown data types

Commit:
Ensure state ignores unknown data types

Commit:
Filter to alarm attributes

Commit:
Filter to alarm attributes

Commit:
Fix formatting

Commit:
Fix formatting

SteveEasley commented:
@tschmidty69 Thanks for testing that! I decided to run out and buy a contact switch to test this. Although I was not getting the same error as you, I think I have fixed your issue. Also, I confirmed the its working as expected (sensor value is refreshed on HA startup).

bakedraccoon committed:
Set tahoma cover scan interval to 60 seconds (#11447)

* Increase tahoma scan interval to 60

* import timedelta

* fix lint

balloob committed:
Add some tests to the cloud component (#11460)

cdce8p committed:
Input Select - Added service description (#11456)

cdce8p committed:
Input Boolean - Deleted 'DEFAULT_INITIAL' (#11453)

cdce8p committed:
Updated gitignore file (#11452)

titilambert committed:
Update hydroquebec component to use hass httpsession (#11412)

* Update hydroquebec component to use hass httpsession

* Remove blank line

Commit:
Catch everything when calling to OctoPrint API to fix #10557 (#11457)

Danielhiversen committed:
Update PULL_REQUEST_TEMPLATE.md (#11465)

* Update PULL_REQUEST_TEMPLATE.md

* Add period

mnoorenberghe committed:
iOS 10 should be served javascript_version:es5 (#11387)

* iOS 10 should be served javascript_version:es5

Fixes #11234

* Update min Safari version to 12

mnoorenberghe committed:
iOS 10 should be served javascript_version:es5 (#11387)

* iOS 10 should be served javascript_version:es5

Fixes #11234

* Update min Safari version to 12

balloob committed:
Version bump to 0.60.1

balloob committed:
Alexa to not use customize for entity config (#11461)

* Alexa to not use customize for entity config

* Test Alexa entity config

* Improve tests

* Fix test

balloob committed:
Merge pull request #11470 from home-assistant/release-0-60-1

0.60.1

balloob committed:
Merge branch 'master' into dev

Chris-V committed:
Don't duplicate html5 registrations (#11451)

* Don't duplicate html5 registrations

If a registration is posted and another registration with the same
endpoint URL exists, update that one instead. That way, we preserve
the device name that has been configured. The previous behavior used to
append 'unnamed device' registrations over and over, leading to
multiple copies of the same registration. The endpoint URL is unique per
service worker so it is safe to update matching registrations.

* Refactor html5 registration view to not write json in the event loop

hawk259 committed:
AlarmDecoder remove icon function as BinarySensorDevice handles it correctly now (#11467)

* remove icon function as BinarySensorDevice handles it correctly now

* removing _type, not used

fabaff committed:
Upgrade psutil to 5.4.3 (#11468)

fabaff committed:
Upgrade alpha_vantage to 1.8.0 (#11476)

balloob committed:
Fix vultr tests (#11477)

fabaff committed:
Upgrade yarl to 0.17.0 (#11478)

abondoe committed:
Add Touchline climate platform (#10547)

* Add toucline platform

* Fix bugs suggested by houndci-bot

* Fix bugs suggested by houndci-bot

* Fix style based on comments and lint

* Remove target temperature

* Fix unexpected EOF

* Fix unexpected EOF

* Fix wrongfully named numberOfDevices variable

* Add target temperature

* Update requirements_all.txt

* Change after review

* Add supported features
* Remove update in constructor
* Set member variables to none in constructor

* Minor changes

fabaff committed:
Upgrade pysnmp to 4.4.4 (#11485)

jesserockz committed:
Add new iGlo component (#11171)

* Add new iGlo component

* Missing comma
Add extra blank lines

* Dont change state in turn_on
Remove unused variables
Update before add

* Fixing some lint issues

fabaff committed:
Add missing configuration variables (#11390)

* Add missing configuration variables

* Minor changes

* Sync platforms and other minor updates

balloob committed:
Clean up Alexa.intent and DialogFlow.intent (#11492)

* Clean up Alexa.intent

* Restructure dialogflow too

* Lint

* Lint

tomwaters committed:
Fix time functions would throw errors in python scripts (#11414)

* Fix time functions would throw errors in python scripts

* Added unit test for time.strptime, change variable name to satisfy lint

* Added docstring for time attribute wrapper method to satisfy lint

* Fixed line too long lint problem

Chris-V committed:
Upgrade pywebpush to 1.5.0 (#11497)

This version includes a fix for the serialization errors that occurred
when updating push subscriptions.

Changes since version 1.3.0:
https://github.com/web-push-libs/pywebpush/compare/1.3.0...28d2b55f

JulianKahnert committed:
upgrade schiene to 0.20 (#11504)

Commit:
timer: include the remaining time in the state attributes (#11510)

Add the amount of remaining time before a timer is finished in its state attributes, so that it is received when fetching a timer state.

In my particular case, it allows me to correctly implement a UI for a timer when it is paused and when the UI didn't received the pause state change (which happens when the UI is started after the pause). In this case, it is impossible to show the remaining amount of time before the timer ends.

Julius2342 committed:
More tolerant KNX component if gateway cant be connected (#11511)

* Issue #11432: Do not stop initializing KNX when tunelling device cant be reached

*  Issue #11432: Mark devices as unavailable if gateway cant be connected

todschmidt committed:
Snips (new) added speech response, parse snips/duration (#11513)

* Snips changed to using <username>:intentName instead of user_IDSTRING__intentName

* added response to snips

* blank line

* added unittests

* houndy

* sdtuff

* more stuff

* Update test_snips.py

* Update snips.py

* Split log tests to avoid dict ordering in py34

* Update test_snips.py

* Update test_snips.py

* still broken

* fixed tests

* fixed tests

* removed fix submitted in another PR

DanNixon committed:
Add Dark Sky weather component (#11435)

amelchio committed:
Lazy loading of service descriptions (#11479)

* Lazy loading of service descriptions

* Fix tests

* Load YAML in executor

* Return a copy of available services to allow mutations

* Remove lint

* Add zha/services.yaml

* Only cache descriptions for known services

* Remove lint

* Remove description loading during service registration

* Remove description parameter from async_register

* Test async_get_all_descriptions

* Remove lint

* Fix typos from multi-edit

* Remove unused arguments

* Remove unused import os

* Remove unused import os, part 2

* Remove unneeded coroutine decorator

* Only use executor for loading files

* Cleanups suggested in review

* Increase test coverage

* Fix races in existing tests

balloob committed:
Fix canary flaky test (#11519)

DanNixon committed:
Address missed review comments for Dark Sky weather (#11520)

See #11435

cpcowart committed:
New features for Owntracks device_tracker (#11480)

* New features for Owntracks device_tracker

- Supporting a mapping of region names in OT to zones in HA, allowing
  separate namespaces in both applications. This is especially helpful
  if using one OT instance to update geofences for multiple homes.
- Creating a setting to ignore all location updates, allowing users to
  rely completely on enter and leave events. I have personally always
  used OT integrations with home automation this way and find it the
  most reliable.
- Allowing the OT topic to be overridden in configuration

* Fixing configuration of MQTT topic, related tests

* Tests for Owntracks events_only feature

* Tests for customizing mqtt topic, region mapping

* Fixing _parse and http for owntracks custom topic

* Making tests more thorough and cleaning up lint

aosadchyy committed:
Reconnect before mochad switch send command (#11296)

* Connection to mochad occasionally stalls on RPi and CM19A. Reconnect one switch send command.

* Formatting and exception hanling fixes

* Moved import inside the method. Logging outside the try-catch.

* Tailing whitespaces.

* MockDependency on pymochad in unit tests to resolve exceptions

* patch pymochad MochadException in unit tests to resolve exceptions

* patch pymochad MochadException in unit tests to resolve exceptions

* cleaned unused import

* lint issue fixed

* pylint issue fixed

florianj1 committed:
Fix Kodi channels media type (#11505)

* Update kodi.py

* Update kodi.py

Fix code style

ChristianKuehnel committed:
Extension sensor alpha_vantage (#11427)

* Extended sensor alpha_vantage

* Improved error handling if symbol is not found. Now we add the symbols that were found,
* Added option to give names and currencies to symbols.
* Added support to read foreign exchange information.
* Icons are selected based on the currency (where MDI has a matching icon)

* added missing line at the end of the file...

* renamed variable as required by pylint

* Fix typos, ordering, and style

Commit:
MQTT json attributes (#11439)

* MQTT json attributes

* Fix lint

* Amends following comments

* Fix lint

* Fix lint

* Add test

* Fix typo

* Amends following comments

* New tests

* Fix lint

* Fix tests

michaelkuty committed:
Support pushing all sensors and fix wrong metrics. (#11159)

For example all metrics with unit % match humidity.
This generate correct metrics like this
# HELP nut_ups_battery_charge sensor.nut_ups_battery_charge
# TYPE nut_ups_battery_charge gauge
nut_ups_battery_charge{entity="sensor.nut_ups_battery_charge",friendly_name="NUT UPS Battery Charge"} 98.0
nut_ups_battery_charge{entity="sensor.nut_ups_battery_charge_2",friendly_name="NUT UPS Battery Charge"} 97.0

yienxu committed:
Add options feature to Baidu TTS. (#11462)

* Add options feature to Baidu TTS.

Add options feature: supported_options() and default_options() added, get_tts_audio updated to accommodate options.

* Add options feature to Baidu TTS.

Add options feature: supported_options() and default_options() added, get_tts_audio updated to accommodate options.

* Fix style

* Fix style

Change the order of content of lists and dictionaries.

* Fix style

Changed order of imports, and fixed grammar errors.

camrun91 committed:
Insteon local update (#11088)

* trying to rework device discovery. now the main component will do the getlinked and pass it to the sub-components. no longer any config needed other than what is needed to connect to the hub. device names are no longer stored. core team told us to stop using configurator to ask for names. there should be a way to set names in hass...possibly this https://home-assistant.io/docs/configuration/customizing-devices/

* fix device types

* make device names just be the isnteon device id

* revert some config changes

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* update insteon client

* linting fixes

* Error Clean up

* Update to make requested changes

* more changes

* Finish requested changes to components

* Fixing Rebase Conflicts

* fix device types

* make device names just be the isnteon device id

* revert some config changes

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* update insteon client

* linting fixes

* Error Clean up

* Update to make requested changes

* more changes

* Finish requested changes to components

* Update Insteon_Local for performance improvements

* Fix errors from get_linked

* Fix typo

* Requested changes

* Fix spacing

* Clean up

* Requested Changes

dshokouhi commented:
Really glad this got submitted, I actually had no idea this was the case with my sensors until I did a reboot last night and saw this PR.   Thanks for getting this done @SteveEasley  :)

todschmidt committed:
Snips: (fix) support new intentName format (#11509)

* support new intentName format

* Added tests for new and old format names

* pylint warning fixes

bacumings commented:
Second that, thanks very much!!

fabaff committed:
Upgrade numpy to 1.14.0 (#11542)

fabaff committed:
Upgrade python-etherscan-api to 0.0.2 (#11535)

fabaff committed:
Upgrade Sphinx to 1.6.6 (#11534)

swilson committed:
ZoneMinder event sensor updates (#11369)

* Switch the ZoneMinder event sensor over to use the consoleEvents API, and add monitored_conditions for 'hour', 'day', 'week', 'month' and 'all'. 'all' is enabled by default to provide backward compatibility with the old behaviour.

* - Switch to new string formatting
- Remove redundant validator

* De-lint

balloob committed:
Move Google Assistant entity config out of customize (#11499)

* Move Google Assistant entity config out of customize

* CONF_ALIAS -> CONF_ALIASES

* Lint

fabaff committed:
Upgrade lightify to 1.0.6.1 (#11545)

cdce8p committed:
Improved test runtime (#11553)

* Remove redundent assert statements and cleanup

* Added 'get_date' function

* Replace 'freeze_time' with 'mock.patch'
* Tox in 185s (py35)

* Removed obsolete 'freeze_time' from test_updater
* Tox 162s (py35)

* Remove test requirement 'freezegun'

* Fixed flake8 errors

* Added 'mock.patch' for 'feedparser.parse'

* Made 'FUNCTION_PATH' a constant

* Remove debug statements.

ericpignet committed:
Add 2 media_player services and 1 custom service to Squeezebox platform (#10969)

* Add 2 media_player services and 1 custom service to Squeezebox platform

* Fix pylint error

* Remove apostrophe in example

* Split method into command and parameters

* Fix Lint error

philk committed:
Fix error on entity_config missing (#11561)

If the `google_assistant` key exists in the config but has no
`entity_config` key under it you'll get an error.

```
  File "/Users/pkates/src/home-assistant/homeassistant/components/google_assistant/http.py", line 51, in is_exposed
    entity_config.get(entity.entity_id, {}).get(CONF_EXPOSE)
AttributeError: 'NoneType' object has no attribute 'get'
```

fabaff committed:
Upgrade keyring to 10.3.2 (#11531)

nkgilley committed:
Coinbase.com sensor platform (#11036)

* coinbase sensors

use hass.data, load_platform

* add exchange rate sensors

dont pass complex object over event bus

* check for auth error

andreacampi committed:
Try to fix crashes after Hue refactoring (#11270)

* Try to fix crashes after Hue refactoring

Refs #11183

* Fix usage of dispatcher_send via helper.

* Address review feedback.

balloob committed:
Remove execution file perm (#11563)

amelchio committed:
Deprecate explicit entity_id in template platforms (#11123)

* Deprecate explicit entity_id in template platforms

* Use config validator for deprecation

* Fix copy/paste typos

* Also print the config value

* Add test for config validator

* Assert the module name that logged the message

balloob committed:
Update frontend to 20180110.0

balloob committed:
Update frontend to 20180110.0

balloob committed:
Test tweaks (#11560)

* Fix is_allowed_path on OS X

* Move APNS2 inside func in test

bakedraccoon committed:
Add Velux Windows to Tahoma (#11538)

* Add Velux Windows to Tahoma

* fix linit

* add supported Tahoma devices by @glpatcern

* hound

* lint

* fix logging

* lint

* lint

* remove blank line after docstring

* changes based on notes by @armills

* fix logging

pvizeli committed:
Extend hass.io services / updater (#11549)

* Extend hass.io services

* Add warning for carfuly options with hass.io

* update tests

* finish tests

* remove update calls

* address comments

* address comments p2

* fix tests

* fix tests

* Use token also for proxy

* Add test for server_host

* Fix test

* Fix tests

* Add test for version

* Address comments

andrey-git committed:
Add sensibo_assume_state service to Sensibo climate (#11567)

randellhodges committed:
Tracking all groups to allow changing of existing groups (#11444)

* Tracking all groups to allow changing of existing groups

* Unit tests

* Fix flake8 warnings in test

amelchio committed:
Fix new squeezebox service descriptions for lazy loading (#11574)

DanNixon committed:
Mark REST binary_sensor unavailable if request fails (#11506)

* Mark REST binary_sensor unavailable if request fails

* Add test suite for RESTful binary sensor

andersonshatch committed:
Tado improvements - hot water zone sensors and climate precision (#11521)

* Add tado hot water zone sensors

* Set precision to match tado app/website

ulido committed:
Make the rpi_rf component thread-safe using an RLock (#11487)

* Make the rpi_rf component thread-safe

The previous implementation suffered from race conditions when two rpi_rf switches are triggered at the same time. This implementation uses an RLock to give one thread at a time exclusive access to the rfdevice object.

* cleanup

* fix lint

danielperna84 committed:
Update pyhomematic, support new devices (#11578)

hawk259 committed:
Alarmdecoder add validation of the zone types (#11488)

* Alarmdecoder add validation of the zone types

* fix line length

emlove committed:
Support OSRAM lights on ZHA (#11522)

swilson committed:
Fix bluetooth tracker source (#11469)

* Add new sources for bluetooth and bluetooth_le trackers (rather than just using the generic 'gps' catch-all).

* Fix lint issues

kennedyshead committed:
Fix for asuswrt, telnet not working and presence-detection for router mode (#11422)

* Fix for telnet mode
Fix for presence detection

closes #11349 #10231

* Optimisation when device is disconnected

balloob commented on homeassistant/components/binary_sensor/zha.py:
```diff
@@ -87,3 +87,31 @@ def cluster_command(self, aps_frame, tsn, command_id, args):
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        @asyncio.coroutine
+        def safe_read(cluster, attributes):
```
Please extract this method and move it to the `zha` component and import it from there. There is no need to copy this method to every platform.

dmulcahey replied:
I may be wrong here but I think `should_poll(self) -> bool:` also needs to be overridden to return false on the base ZHA entity class. With debugging on, it appears that hass is calling the `async_update` functions that were added to these classes to poll for changes behind the scenes. The zigbee devices are already configured to push changes to hass when necessary to preserve battery life. I think leaving this as is will severly affect battery life of the sensors. @balloob, does this make sense? or am I way off base?

andrey-git committed:
Remove aux_heat support from Sensibo now that UI supports on/off (#11579)

Laqoore committed:
Changed device type of media player and cover to switch (#11483)

* Changed device type of media player and cover to switch

Covers and media players should not be of device type 'light'. Example: If user requests all lights to switch to off, covers are closed and media players are affected too.

* Fix test

rwa committed:
Control ordering of display in history component (#11340)

* Make the order shown in the history component match the ordering given in the configuration of included entities (if any)

* return the sorted result

* optimize sorting.  since entities only appear once, we can break from a search on first find, and no copy of the list is needed

balloob committed:
Pr/11430 (#11587)

* Fix error when name is non-latin script

When the name is non-latin script (e.g. Japanese), slugify returns empty string and causes failure in a later stage. This commit fixes the issue by using default name.

* Add test

cgtobi committed:
Cast return values and add unit tests for the yahoo weather component. (#10699)

* Add unittest for yahoo weather component.

* Add requitements for tests.

* Update requirements.

* Revert "Update requirements."

This reverts commit 024b5cf8bf0d1bdfe927a117f5befb53c5a5cff4.

* Update requirements.

* Randomize sample data.

* Remove unnecessary methods.

* Remove dependency and replace with MockDependency.

* Add temporary fix for yahoo weather API issue.

bakedraccoon committed:
add velux roller shutter to tahoma (#11586)

With #11538 I forgot to add another type of Roller shutters that should be supported.

abjorshammar committed:
Added support for enable/disable motion detection (#11583)

* Added support for enable/disable motion detection

* Changed name of variable for exception, lint error

* Moved motion_detection_enabled property to the other properties

balloob committed:
Update frontend

jackwindows committed:
Avoid returning empty media_image_url string (#11557)

This relates to issue
https://github.com/home-assistant/home-assistant/issues/11556

jackwindows committed:
Avoid returning empty media_image_url string (#11557)

This relates to issue
https://github.com/home-assistant/home-assistant/issues/11556

emlove committed:
Fix state for trigger with forced updates (#11595)

emlove committed:
Fix state for trigger with forced updates (#11595)

rwa committed:
Concord232 alarm arm away fix (#11597)

* fix arming away cmd for concord232 client

* bump required version of concord232 to 0.15

rwa committed:
Concord232 alarm arm away fix (#11597)

* fix arming away cmd for concord232 client

* bump required version of concord232 to 0.15

pvizeli committed:
Core support for hass.io calls & Bugfix check_config (#11571)

* Initial overwrites

* Add check_config function.

* Update hassio.py

* Address comments

* add hassio support

* add more tests

* revert core changes

* Address check_config

* Address comment with api_bool

* Bugfix check_config

* Update core.py

* Update test_core.py

* Update config.py

* Update hassio.py

* Update config.py

* Update test_config.py

pvizeli committed:
Core support for hass.io calls & Bugfix check_config (#11571)

* Initial overwrites

* Add check_config function.

* Update hassio.py

* Address comments

* add hassio support

* add more tests

* revert core changes

* Address check_config

* Address comment with api_bool

* Bugfix check_config

* Update core.py

* Update test_core.py

* Update config.py

* Update hassio.py

* Update config.py

* Update test_config.py

bakedraccoon committed:
Fix Tahoma stop command for 2 types of shutters (#11588)

* add working stop command

This fixes the stop command for 2 types of roller shutters

* fix line too long

* fix indentation

* fix indentation

bakedraccoon committed:
Fix Tahoma stop command for 2 types of shutters (#11588)

* add working stop command

This fixes the stop command for 2 types of roller shutters

* fix line too long

* fix indentation

* fix indentation

todschmidt committed:
Snips add say and say_actions services (new) (#11596)

* Added snips.say and snips.say_action services

* Added snips.say and snips.say_action services

* Merged services.yaml changes I missed

* added tests for new service configs

* Woof

* Woof Woof

* Changed attribute names to follow hass standards.

* updated test_snips with new attribute names

todschmidt committed:
Snips add say and say_actions services (new) (#11596)

* Added snips.say and snips.say_action services

* Added snips.say and snips.say_action services

* Merged services.yaml changes I missed

* added tests for new service configs

* Woof

* Woof Woof

* Changed attribute names to follow hass standards.

* updated test_snips with new attribute names

todschmidt committed:
Pushbullet email support (fix) (#11590)

* Simplified push calls

* Cleaned up and added unittests

* Fixed email parameter

* Fixed email parameter

todschmidt committed:
Pushbullet email support (fix) (#11590)

* Simplified push calls

* Cleaned up and added unittests

* Fixed email parameter

* Fixed email parameter

Danielhiversen committed:
Bugfix and cleanup for Rfxtrx (#11600)

* rfxtrx clean up

* rfxtrx clean up

* rfxtrx clean up

Danielhiversen committed:
Bugfix and cleanup for Rfxtrx (#11600)

* rfxtrx clean up

* rfxtrx clean up

* rfxtrx clean up

Danielhiversen committed:
Xiaomi lib upgrade (#11603)

* upgrade xiaomi lib

* xiaomi lib

Danielhiversen committed:
Xiaomi lib upgrade (#11603)

* upgrade xiaomi lib

* xiaomi lib

fabaff committed:
Upgrade yarl to 0.18.0 (#11609)

fabaff committed:
Upgrade yarl to 0.18.0 (#11609)

fabaff committed:
Upgrade keyring to 10.6.0 (#11608)

fabaff committed:
Upgrad youtube_dl to 2017.12.31 (#11610)

bakedraccoon committed:
patch stop command (#11612)

bakedraccoon committed:
patch stop command (#11612)

pvizeli committed:
small sonos cleanup (#11607)

keesschollaart81 committed:
Update Xiaomi Miio compontent broken URI (#11621)

jkoelker committed:
light/mqqt_json: allow brightness scaling (#11613)

nickovs committed:
Adding support for Lutron covers (#11602)

* Adding support for Lutron Radio RA2 shades as cover components.

* Adding initial version of the Lutron shades component.

* Fixed line-length violation detected by Hound.

arsaboo committed:
Update Pyarlo to 0.1.2 (#11626)

arsaboo committed:
Update Pyarlo to 0.1.2 (#11626)

jesserockz committed:
Use kelvin/mireds correctly for setting iglo white (#11622)

* Use kelvin/mireds correctly for setting iglo white

* Update requirements_all.txt

* Fix line lengths

jesserockz committed:
Use kelvin/mireds correctly for setting iglo white (#11622)

* Use kelvin/mireds correctly for setting iglo white

* Update requirements_all.txt

* Fix line lengths

starkillerOG committed:
Hyperion: Add brightness, HDMI and effect support (#11543)

* Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.

* Hyperion: Add brightness, HDMI and effect support

- added brightness support to dim the hyperion light
- changed the "OFF" command to set the color to [0,0,0] after clearing all priorities.
  This is neccesary to keep the light turned off when an HDMI grabber is used for ambilight with hyperion.
- added HDMI ambilight mode recognition and control.
  by setting the "hdmi_priority" in your "configuration.yaml" file (defaults to 880), home assistant will now be able to recognize when the hyperion light is in HDMI ambilight mode and will change its icon to an HDMI symbol and set the status to ON.
Switching the hyperion light to HDMI ambilight mode can be done through the effect option (clears all priorities such that the HDMI grabber remains).
- added effect support for the default effects of hyperion, a custom list can be defined in the "configuration.yaml" file by using the "effect_list" option.
- fixed some style issues with too long lines

* Hyperion: Add brightness, HDMI and effect support 

 - fixed some more indentation style issues

* Hyperion: Add brightness, HDMI and effect support 

- yet more fixed visuel indent issues

* Hyperion: Add brightness, HDMI and effect support 

- more visuel indents

* Hyperion: Add brightness, HDMI and effect support

- fixed invalid variable "A"

* Hyperion: Add brightness, HDMI and effect support

- remove unnececary brackets
- specify specific exceptions

* correct changing state holding attributes during a service method

Proccesed the comments of @MartinHjelmare: https://github.com/home-assistant/home-assistant/pull/11543#pullrequestreview-88328659

* indent correction

corrected tab instead of 4 spaces

* Hyperion: Add brightness, HDMI and effect support 

- changed 'none' to None
- renamed "self._skip_check" to "self._skip_update"

* Add brightness, HDMI and effect support 

changed checking if a list is empty from "list == []" to "not list"

glpatcern committed:
Fixes and enhancements for the Tahoma platform (#11547)

* Strip off the RTS/IO ID from the entity ID

* Ignore exception thrown when the device does not provide an active state

* Send actions with a label for easier identification in the Tahoma log

* Linting

* Strip off the RTS/IO ID from the entity ID, take 2

As per suggestions, let HA do the standard initialization and assign
an appropriate entity_id, instead of overriding it with the lengthy
unique_id coming from the TaHoma devices.

Danielhiversen committed:
upgrade xiaomi lib (#11629)

tinloaf committed:
Add templates to MQTT climate (#11623)

* Add templates

* Make pylint happy

pvizeli committed:
Update CODEOWNERS

fabaff committed:
Upgrade coinmarketcap to 4.1.2 (#11634)

fabaff committed:
Merge pull request #11589 from home-assistant/release-0-61

0.61

fabaff committed:
Merge branch 'master' into dev

Kane610 committed:
Add Deconz support for Zigbee green power devices like Hue Tap (#11455)

* Add support for Zigbee green power devices such as the Hue Tap

* Clarify that imported SWITCH is a DECONZ_REMOTE

fabaff committed:
Bump dev to 0.62.0.dev0 (#11652)

Commit:
Merge branch 'zha-switch-state' of github.com:SteveEasley/home-assistant into zha-switch-state

SteveEasley commented:
Wow my rebase of upstream dev went very wrong not sure what happen (git is hard). Going to just close this PR and recreate.

dmulcahey commented:
@SteveEasley I started incorporating your changes [here](https://github.com/dmulcahey/home-assistant/commit/964f71797ad3c4a0f9d78baab01bd30a8ba76bc4) We can work together on this if you like? Or I can pull the changes from your PR out and then incorporate them when you're ready.

SteveEasley commented:
@dmulcahey I just saw this after creating https://github.com/home-assistant/home-assistant/pull/11672. Thats fine if you want to pull them into your PR. Let me know and I will close this one if so.

Side note: I did turn off polling as you suggested for now. Although it does have the side effect of on my SmartThings switch outlet of not updating the ha state if I manually push the on/off switch on the device itself. After digging a little, the outlet unfortunately does not send a wireless update that the state changes to the controller. Though my change does at least read the state on HA start.  I am on the fence about disabling polling. I get your point about battery life, but wonder if thats a factor. If a device is asleep, it wont receive the poll request anyway (or at least the controller may buffer it till the end device wakes). I would play with it more, but there are some race conditions causing weird log entries that I want to explore when I get time (these existed before my changes). Errors like `Unexpected message send notification`. I think those are errors might be causing HA to get the updates we want.

dmulcahey commented:
I’ll let your new one go through. Then I’ll submit mine. 

I am using a Smartthings plug as well on my test setup. I had to make a change to the zha file in the switch directory to correct the updating issue you mentioned... check it out for details.

As for the polling... I don’t think it’s even necessary. The switch thing isn’t a polling issue... the polling could keep devices that should sleep awake if the interval is fast enough which would kill battery powered devices much faster than normal.

SteveEasley commented:
>  I had to make a change to the zha file in the switch directory to correct the updating issue you mentioned

Odd, my PR has the same changes you have in switch/zha.py. Are you saying on your test setup you can push the button on the plug itself, and HA will update eventually to the proper state eventually?

dmulcahey commented:
Ah, missed “on the device itself” I’ll try to mess with that as well. It may need reporting configured on the state.

SteveEasley commented:
I added some debugging to `bellows.ezsp.EZSP::frame_received()` and never saw a packet come in when I pushed the button. So I don't think there is a solution (for real time anyway).

dmulcahey commented:
All set. Take a look at my switch class again. It just needed attribute reporting configured on the on/off cluster for the onoff attribute.

SteveEasley commented:
Nice! Confirmed on my end too. Want me to roll that switch/zha.py change into my PR since its related?

dmulcahey commented:
Go for it
